lib/rubocop/cop/chef/modernize/allowed_actions_initializer.rb

```diff
@@ -0,0 +1,76 @@
+#
+# Copyright:: 2019, Chef Software Inc.
+# Author:: Tim Smith (<tsmith@chef.io>)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+module RuboCop
+  module Cop
+    module Chef
+      module ChefModernize
+        # The allowed actions can now be specified using the `allowed_actions` helper instead of using the @actions or @allowed_actionsn variables in the resource provider initialize method. In general we recommend against writing HWRPs, but if HWRPs are necessary you should utilize as much of the resource DSL as possible.
+        #
+        # @example
+        #
+        #   # bad
+        #   def initialize(*args)
+        #     super
+        #     @actions = [ :create, :add ]
+        #   end
+        #
+        #   # also bad
+        #   def initialize(*args)
+        #     super
+        #     @allowed_actions = [ :create, :add ]
+        #   end
+        #
+        #   # good
+        #   allowed_actions [ :create, :add ]
+        #
+        class AllowedActionsFromInitialize < Cop
+          include RangeHelp
+
+          MSG = 'The allowed actions of a resource can be set with the "allowed_actions" helper instead of using the initialize method.'.freeze
+
+          def on_def(node)
+            return unless node.method_name == :initialize
```

Comment:
There's no reasonable case where someone might define these outside of `initialize` is there?

Reply:
If they're using them outside the intialize that's a different case and we want to avoid breaking their code. It could just be any old library setting those values and fixing that would be bad.